src/crypto/keys.ts

```diff
@@ -1,23 +1,24 @@
-import {KeyPairState, Wallet} from "../interfaces";
-import {fromHexString} from "../util/hex";
-import nacl from "tweetnacl";
-
-// Generates ed25519 public/private keypair from seed
-function generateKeyPairFromSeed(seedString: string): KeyPairState {
-  if (seedString.length < 32) { throw new Error("Invalid seed string."); }
-  const seed = fromHexString(seedString.substr(0, 64));
-  return nacl.sign.keyPair.fromSeed(seed);
-}
+import {AccountState, Wallet} from "../interfaces";
+import { Keyring } from '@polkadot/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import { stringToU8a } from '@polkadot/util';
 
 // Generate keypair from metamask wallet interface using app key
-export async function generateKeys(wallet: Wallet): Promise<KeyPairState> {
-  const appKey = await wallet.getAppKey();
-  const keypair = generateKeyPairFromSeed(appKey);
+export async function generateKeys(wallet: Wallet): Promise<KeyringPair> {
+  // get app key and wait for api to be ready
+  const result = await Promise.all([wallet.getAppKey(), cryptoWaitReady()]);
+  const appKey = result[0];
+  // generate keys
+  const seed = (appKey.substr(0, 32));
+  const keyring = new Keyring();
+  const keyringPair = keyring.addFromSeed(stringToU8a(seed));
+  const accountState = {keyring: keyringPair.toJson(), seed} as AccountState;
```

Comment:
const accountState: AccountState so you don't cast

Reply:
why do you store seed?
We should probably encrypt keyring with seed so attacker cannot get plaintext account from local storage

Reply:
@mpetrunic everything else is finished, I think this should be moved to new task (**encrypt keyring**).